### PR TITLE
Fully convolutional network with test time augmentation

### PR DESCRIPTION
**Pull request by StefanGerlach**
This update introduces a fully convolutional neuronal network for fast traffic light detection. To detect traffic lights, the activation maps for each class are activated with softmax (per pixel) and compared. 

High probabilites of traffic light detections on the activation map are accepted for a valid detection of a traffic light. The confidence thresholds are configured per traffic light state.

To make the system more robust, test time augmentation (horizontal flip) is used. 

This is tested on 5 udacity rosbags and runs quite well, but "Green" is a little bit to sensitive.

### CODE REVIEW

File: ros/src/tl_detector/light_classification/tl_classifier.py [hunk 1]
```diff
@@ -19,47 +19,25 @@ class TLClassifier(object):
...
-        with self.session.graph.as_default():
-            od_graph_def = tf.GraphDef()
-            with tf.gfile.GFile(path_to_tensorflow_graph, 'rb') as fid:
-                serialized_graph = fid.read()
-                od_graph_def.ParseFromString(serialized_graph)
-                tf.import_graph_def(od_graph_def, name='')
-
-                g = tf.get_default_graph()
+        with self.session.as_default():
+            saver = tf.train.import_meta_graph(path_to_tensorflow_graph+'/model.ckpt.meta')
```
Comment (stela) [inline ros/src/tl_detector/light_classification/tl_classifier.py:29]: Missing the model.ckpt.meta file, and the old frozen_inference_graph.pb could then be deleted I guess?

Comment (StefanGerlach) [reply ros/src/tl_detector/light_classification/tl_classifier.py:29]: I missed the git add for that! Thanks for seeing this.

Comment (stela) [reply ros/src/tl_detector/light_classification/tl_classifier.py:29]: The sim config says:
`tensorflow_graph_path: 'light_classification/inference_graph/sim/frozen_inference_graph.pb'`
whereas the site config says:
`tensorflow_graph_path: 'light_classification/fcn_graph/checkpoint'`
Will both the path to a directory containing the files as well as full path to a .pb file work with this? Looks like the simulator config would fail?

File: ros/src/tl_detector/site_traffic_light_config.yaml [hunk 2]
```diff
@@ -4,15 +4,17 @@ camera_info:
...
-tensorflow_confidence_thesh: 0.6
-detections_filter_size: 3
-no_detections_filter_size: 12
-no_detections_threshold: 0.8
-state_count_threshold: 2
+use_tensorflow_nth_frame: 2
+tensorflow_graph_path: 'light_classification/fcn_graph/checkpoint'
+tensorflow_confidence_thesh_green: 0.85
+tensorflow_confidence_thesh_yellow: 0.85
+tensorflow_confidence_thesh_red: 0.85
```
Comment (stela) [inline ros/src/tl_detector/site_traffic_light_config.yaml:17]: sim config still uses the old single setting `tensorflow_confidence_thesh`, not split by RGB, (besides pointing to a .pb file instead of a directory).

**Comment by stela** [comment]
Looks good now, just need to test it with the simulator (long track + camera-less churchlot/site) and ROS bags.
